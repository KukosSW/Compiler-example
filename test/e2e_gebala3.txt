DECLARE
    n, m, r, p, d
BEGIN

    READ n;
    d := 2;
    m := d * d;

    WHILE n >= m DO
        p := 0;
        r := n % d;
        WHILE r = 0 DO
            n := n / d;
            p := p + 1;
            r := n % d;
        ENDWHILE

        IF p >0 THEN
            WRITE d;
            WRITE p;
        ELSE
            d := d + 1;
            m := d * d;
        ENDIF
    ENDWHILE

    IF n != 1 THEN
        WRITE n;
        WRITE 1;
    ENDIF

END