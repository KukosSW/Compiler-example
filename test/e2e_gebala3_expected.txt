2
1
3
2
47
1
14593
1
